rrs/views.py

```diff
@@ -0,0 +1,1005 @@
+                context['upstream_version'] = recipe_upstream.version
+                context['upstream_no_update_reason'] = recipe_upstream.no_update_reason
+
+        self.recipe_maintainer_history = RecipeMaintainerHistory.get_last(recipesymbol.layerbranch)
+        recipe_maintainer = RecipeMaintainer.objects.filter(recipesymbol=recipesymbol,
+                history=self.recipe_maintainer_history)
+        if recipe_maintainer:
+            maintainer = recipe_maintainer[0].maintainer
+            context['maintainer_name'] = maintainer.name
+        else:
+            context['maintainer_name'] = 'No maintainer'
+
+        details = []
+        multigroup = False
+        lastgroup = '' # can't use None here
+        for ru in RecipeUpgrade.objects.filter(recipesymbol=recipesymbol).exclude(upgrade_type='M').order_by('group', '-commit_date', '-id'):
+            details.append(_get_recipe_upgrade_detail(maintplan, ru))
+            if not multigroup:
+                if lastgroup == '':
+                    lastgroup = ru.group
+                elif ru.group != lastgroup:
+                    multigroup = True
+        details.sort(key=lambda s: RecipeUpgradeGroupSortItem(s.group), reverse=True)
+        context['multigroup'] = multigroup
+        context['recipe_upgrade_details'] = details
+        context['recipe_upgrade_detail_count'] = len(details)
+
+
+        if not recipe:
+            ru = RecipeUpgrade.objects.filter(recipesymbol=recipesymbol).order_by('-commit_date', '-id').first()
+            if ru:
+                context['last_filepath'] = ru.filepath
+
+        context['recipe_layer_branch_url'] = _get_layer_branch_url(
+                recipesymbol.layerbranch.branch.name, recipesymbol.layerbranch.layer.name)
+
+        context['recipe_provides'] = []
+        if recipe:
+            for p in recipe.provides.split():
+                context['recipe_provides'].append(p)
+
+            context['recipe_depends'] = StaticBuildDep.objects.filter(recipes__id=recipe.id).values_list('name', flat=True)
+
+            context['recipe_distros'] = RecipeDistro.get_distros_by_recipe(recipe)
+        else:
+            context['recipe_depends'] = []
+            context['recipe_distros'] = []
+
+        context['otherbranch_recipes'] = Recipe.objects.filter(layerbranch__layer=recipesymbol.layerbranch.layer, layerbranch__branch__comparison=False, pn=recipesymbol.pn).order_by('layerbranch__branch__sort_priority')
+
+        return context
+
+class MaintainerList():
+    name = None
+    recipes_all = 0
+    recipes_up_to_date = '0'
+    recipes_not_updated = '0'
+    recipes_cant_be_updated = '0'
+    recipes_unknown = '0'
+    percentage_done = '0.00'
+
+    interval_statistics = None
+
+    def __init__(self, name):
+        self.name = name
+
+class MaintainerListView(ListView):
+    context_object_name = 'maintainer_list'
+
+    def get_queryset(self):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 16 other functions. By comparison, 98% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this function or break it into multiple functions.